### ISSUE TITLE
[Client] [Authorization] [Oauth] client.get_dataset_versions does not support Authentication

### ISSUE THREAD

**Issue by carlosmagan** [Arize-ai/phoenix#4131]
**Describe the bug**
The client.get_dataset_versions method does not handle authentication properly.

In this method we have a plain httpx request that does not include headers, therefore authorization is not supported returning a 401 Unauthorized.
`response = httpx.get(url=url, params={"limit": limit})`
This method should use the method self._client.get in an analogous way to the get_dataset method
```
response = self._client.get(
            urljoin(self._base_url, f"v1/datasets/{dataset_id}/versions"),
            params={"limit": limit},
        )
```

**To Reproduce**
1. Deploy phoenix on a secure environment that requires Oauth authentication
2. Upload a dataset
3. Instantiate a phoenix client in a notebook
4. Query the dataset with the get_dataset_versions method

**Expected behavior**
get_dataset_versions should handle authorization properly, returning the dataset versions when the client uses Oauth authentication.

**Screenshots**
Failed behavior:
![image](https://github.com/user-attachments/assets/7bd19713-c06a-40d6-90d5-655f093d5018)
Expected behavior:
![image](https://github.com/user-attachments/assets/124018f9-a2e1-47fd-8aa7-cc26e3072d94)

**Comment by axiomofjoy** [comment]
@carlosmagan Thanks for using Phoenix. We hear you on the need for authentication and are beginning work on this [here](https://github.com/Arize-ai/phoenix/issues/3433).

`phoenix.Client` does not currently support OAuth2, so I am relabeling this as an enhancement request rather than a bug. Please let us know if you have any suggestions for auth-related features you'd like to see.

**Comment by axiomofjoy** [comment]
Authentication, including support for OAuth2, was included in Phoenix 5.0. Please see the docs [here](https://docs.arize.com/phoenix/deployment/authentication).
